Changes C4 type-path

nullbear opened the pull request:
Changes c4 path from obj/item/weapon/c4 to
obj/item/weapon/grenade/plastic/c4, removing placeholder and removing the extra paths and providing potential for a lot more functionality. Some maps may need to be updated.

added whitelist/blacklist for c4 objects. Generally, if an object uses overlays.cut(), it should be added to the blacklist.

Removes duplicate explosive lance recipe, now that c4 is a proper
subtype of grenades it is not necessary.

phil235 commented:
Adding a isPXproof var to every atom/movable is bad.

What's the goal of this PR?

ChangelingRain commented on code/modules/mob/living/simple_animal/friendly/dog.dm:
```diff
@@ -159,7 +159,7 @@
 						return
-					if(istype(item_to_add,/obj/item/weapon/c4)) // last thing he ever wears, I guess
+					if(istype(item_to_add,/obj/item/weapon/grenade/plastic/c4)) // last thing he ever wears, I guess
 						item_to_add.afterattack(src,usr,1)
```
This doesn't even woooooork, there's no mob override so it won't place the c4 on the dog

nullbear replied:
The mob override is on the dog itself. Like it always was.

ChangelingRain commented on code/game/objects/items/weapons/grenades/plastic.dm:
```diff
@@ -1,115 +1,109 @@
+	message_admins("[key_name_admin(user)](<A HREF='?_src_=holder;adminmoreinfo=\ref[user]'>?</A>) (<A HREF='?_src_=holder;adminplayerobservefollow=\ref[user]'>FLW</A>) suicided with [src.name] at ([x],[y],[z] - <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[x];Y=[y];Z=[z]'>JMP</a>)",0,1)
+	message_admins("[key_name(user)] suicided with [src.name] at ([x],[y],[z])")
+	sleep(10)
+	prime(get_turf(user))
```
This arg doesn't do anything?

optimumtact commented:
also did you properly mapmerge z2 or is it being converted?

nullbear commented:
@phil235 changed that. Uses a blacklist/whitelist instead. @ChangelingRain you can give ian that new hat properly now. @optimumtact Mapmerger's been busted since 510 i thought? I used notepad++.

Core0verload commented:
useless

PKPenguin321 commented:
for what purpose

optimumtact commented:
>mapmergers been busted since 510

AYY WHAT

nullbear commented:
@core0verload It gets rid of the duplicate spear recipe, allows things to use c4 + grenade code, is a minor change, and i intend to make use of it in a later pr.

Fox-McCloud commented:
whyyy

nullbear commented:
@Fox-McCloud why would the alternative be better? This is a minor change with few benefits, but still has benefits over the old path.

nullbear committed:
Updates to use new c4 path

nullbear commented:
Alright i've made a few tweaks. 
I've left the old c4 path in there as a depreciated version for the time being. And removed all map edits.
I've added a c4 variant, to demonstrate potential uses for the new path.

phil235 commented on code/game/objects/items/weapons/grenades/plastic.dm:
```diff
@@ -0,0 +1,147 @@
+	if (!flag)
+		return
+	if ((ishuman(AM) || AM in pxblacklist) && !AM in pxwhitelist)
+		return
```
the whole blacklist/whitelist thing won't be needed after #17092 is merged. At least, if you correctly make plastic/attackby() end with `return ..()` instead of `..()`.

nullbear replied:
alright, i'll change it after 17092 is merged.

phil235 commented on code/game/objects/items/weapons/explosives.dm:
```diff
@@ -51,13 +48,7 @@
-		wires.interact(user)
-	else
-		..()
+	..()
```
The proc is empty, so just remove the whole proc.

nullbear replied:
c4 can't be defused anyways. This is only used for attaching signallers.

phil235 commented on code/game/objects/items/weapons/grenades/plastic.dm:
```diff
@@ -0,0 +1,153 @@
+	else if(is_wire_tool(I))
+		wires.interact(user)
+	else
+		..()
```
return ..()

nullbear committed:
Squish

nullbear committed:
fuck around
